src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -85,13 +85,20 @@ define(function (require, exports, module) {
                 }
 
                 var isDir = file.options.dir;
+                var filename = removeProjectFolder(file.name);
                 filenames.push({
-                    absPath: Path.join(destination, file.name),
+                    absPath: Path.join(destination, filename),
                     isDirectory: isDir,
                     data: isDir ? null : new Buffer(file.asArrayBuffer())
                 });
             });
 
+            function removeProjectFolder(path){
+                // Nuke root folder thimble-project/ if exists
```

Comment:
Expand this comment slightly:

```js
// Nuke root folder `thimble-project/` if exists so that project zip files
// can be re-imported without adding an unnecessary folder.
```